saw-central/src/SAWCentral/Builtins.hs

```diff
@@ -1369,35 +1369,37 @@ proveByBVInduction script t =
             --
             --   ((x : Vec w Bool) -> ((y: Vec w Bool) -> is_bvult w y x -> p y) -> p x)
             thmHyp <- io $
-                do vars  <- reverse <$> mapM (scLocalVar sc) [ 0 .. length pis - 1]
+                do vars  <- mapM (scExtCns sc) pis
                    t1    <- scApplyAllBeta sc (ttTerm t) vars
                    tsz   <- scTupleSelector sc t1 1 2 -- left element
                    tbody <- scEqTrue sc =<< scTupleSelector sc t1 2 2 -- rightmost tuple element
-                   tsz_shft   <- incVars sc 0 (length pis) tsz
+                   ivars <- sequence [ scFreshGlobal sc ("i_" <> toShortName (ecName ec)) (ecType ec) | ec <- pis ]
```

Comment:
Is it worth factoring out the `ecs -> sequence [ scFreshGlobal sc ("i_" <> toShortName (ecName ec)) (ecType ec) | ec <- ecs ]` bit into its own local definition? It's used both here and below in the definition of `indHypProof`.

Reply:
In 226b2df I factored out that and several other reused subterms.